### PR TITLE
feat(@angular-devkit/build-angular): add option to retain CSS special comments in global styles

### PR DESCRIPTION
**Pull request by alan-agius4**
Prior to this change special CSS comments `/*! comment */` were being removed during minification when using the application builder. This caused tools that ran post build that rely on such comments such as purgeCSS and critters not to function properly.

We now provide a `removeSpecialComments` option to enable retention of these comments in global CSS files.

Usage example:
```json
{
  "projects": {
    "my-app": {
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:application",
          "configurations": {
            "production": {
              "optimization": {
                "styles": {
                  "removeSpecialComments": false
                }
              }
            }
          }
        }
      }
    }
  }
}
```

Closes: #26432

### CODE REVIEW

**Comment by CarlosTorrecillas** [comment]
Thanks @alan-agius4, as always, providing an outstanding turnaround !

**Comment by CarlosTorrecillas** [comment]
Hi guys, just for our information, is this planned to be merged at some point soon? No rush at all, it is because this is holding us back from release the upgrade to Angular 17. Thanks!

**Comment by alan-agius4** [comment]
Had a chat with @clydin earlier last week and we decided to go with an additional option. The PR has been updated to add this option.

This should be available as part of 17.1.0.

**Review by clydin — APPROVED** [review]

File: packages/angular_devkit/build_angular/src/builders/application/tests/options/optimization-remove-special-comments_spec.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        },
+      });
+
+      const { result } = await harness.executeOnce();
+      expect(result?.success).toBeTrue();
+
+      harness.expectFile('dist/browser/styles.css').content.not.toContain('important-comment');
+    });
+
+    it(`should not retain special comments when 'removeSpecialComments' is set not set`, async () => {
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/builders/application/tests/options/optimization-remove-special-comments_spec.ts:63]:
```suggestion
    it(`should not retain special comments when 'removeSpecialComments' is not set`, async () => {
```

**Pull request merged**